Translator/Extractor/JMSTranslatorBundle/AdminExtractor.php

```diff
@@ -66,12 +71,18 @@ class AdminExtractor implements ExtractorInterface, TranslatorInterface, Securit
      */
     public function __construct(Pool $adminPool, LoggerInterface $logger = null)
     {
+        @trigger_error(
+            'Implementing the Symfony\Component\Translation\TranslatorInterface for '.__CLASS__
+            .' is deprecated since version 3.x and will be removed in 4.0.',
+            E_USER_DEPRECATED
+        );
```

Comment:
Don't trigger errors inconditionally please: if you introduce a deprecation, you must provide a way to work around it.